### PR TITLE
Add autoui refscheme support

### PR DESCRIPTION
**Pull request by JSReds**
Change-type: minor
Signed-off-by: Andrea Rosci <andrear@balena.io>

---

##### Contributor checklist

- [ ] I have regenerated jest snapshots for any affected components with `npm run generate-snapshots`

##### Reviewer Guidelines

- When submitting a review, please pick:
  - '_Approve_' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '_Request Changes_' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '_Comment_' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

---

### CODE REVIEW

File: src/extra/AutoUI/models/helpers.ts [hunk 1]
```diff
@@ -59,6 +74,33 @@ export const autoUIRunTransformers = <
...
+
+	const descriptionObject = JSON.parse(property.description!);
+	const refScheme = descriptionObject['x-ref-scheme'];
+	if (Array.isArray(field) && field[0] && !refScheme.length) {
+		return field[0] ?? null;
+	}
+	if (Array.isArray(field) && field[0]) {
+		return get(field[0], refScheme) ?? null;
+	}
+	return get(field, refScheme) ?? null;
```
Comment (thgreasi) [inline src/extra/AutoUI/models/helpers.ts:101]:
```suggestion
	const transformed = (Array.isArray(field) && field.length <= 1 ? field[0] : field) ?? null;
	if (refScheme) {
		return get(transformed, refScheme) ?? null
	}
	return return transformed;
```

File: src/extra/AutoUI/index.tsx [hunk 2]
```diff
@@ -117,6 +118,11 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 		return modelRaw;
 	}, [modelRaw]);
 
+	// const data = React.useMemo(() => {
+	// 	console.log('start!');
+	// 	return autoUIAdaptRefScheme(data, model.schema.properties);
+	// }, [data]);
```
Comment (thgreasi) [inline src/extra/AutoUI/index.tsx:124]: Comment :police_car:

**Review by thgreasi — COMMENTED** [review]
LGTM :tada:

**Comment by JSReds** [comment]
@balena-ci rebase

**Comment by JSReds** [comment]
@balena-ci I self-certify!

**Pull request merged**